include/golioth/ota.h

```diff
@@ -68,6 +68,13 @@ struct golioth_ota_component
     int32_t size;
     /// True, if the component is compressed and requires decompression
     bool is_compressed;
+    /// Artifact Hash
+    char hash[CONFIG_GOLIOTH_OTA_MAX_COMPONENT_HASH_LEN + 1];
+    /// Artifact uri (e.g. "/.u/c/main@1.2.3")
+    char uri [CONFIG_GOLIOTH_OTA_MAX_COMPONENT_URI_LEN + 1];
+    /// Artifact bootloader (e.g. "mcuboot")
+    char bootloader[CONFIG_GOLIOTH_OTA_MAX_COMPONENT_BOOTLOADER_NAME_LEN + 1];
```

Comment:
Adding these fields will increase RAM requirements non-trivially. As part of this PR, let's also change the default value for `CONFIG_GOLIOTH_OTA_MAX_NUM_COMPONENTS` to 1 (currently set at 4, which is excessive imo). Users can increase this config option if they need to do multi-component updates.